File: 692-top-k-frequent-words/top-k-frequent-words.py
class Solution(object):
    def topKFrequent(self, words, k):
        
        map = {}

        for key in words:
            if key in map:
                map[key] += 1
            else:
                map[key] = 1

        ans = []

        while k > 0:
            max_freq = -1
            temp = ""

            for key in map.keys():
                current_freq = map[key]

                if current_freq > max_freq:
                    temp = key
                    max_freq = current_freq
                
                elif current_freq == max_freq:
                    if key < temp:
                        temp = key
            
            ans.append(temp)
            map.pop(temp)
            k -= 1
        
        return ans

File: 692-top-k-frequent-words/top-k-frequent-words.java
class Solution {
    public List<String> topKFrequent(String[] words, int k) {
        HashMap<String, Integer> map = new HashMap<>();
        List<String> ans = new ArrayList<>();

        for (String ch : words) {
            if (map.containsKey(ch)) {
                map.put(ch, map.get(ch) + 1);
            } else {
                map.put(ch, 1);
            }
        }

        for (int i = 0; i < k; i++) {
            int maxfreq = 0;
            String temp = "";
            for (String key : map.keySet()) {
                int currentFreq = map.get(key);
                if (currentFreq > maxfreq) {
                    maxfreq = map.get(key);
                    temp = key;
                }
                else if (currentFreq == maxfreq) {
                    if (temp.isEmpty() || key.compareTo(temp) < 0) {
                        temp = key;
                    }
                }
            }
            ans.add(temp);
            map.remove(temp);
        }
        return ans;
    }
}